### PR TITLE
Add v4.1 schema to mssql

### LINKED ISSUE #4547: Upgrade to Task 2.3.3
Depends on https://github.com/spring-cloud/spring-cloud-task/pull/783.

Addresses an issue reported by users using Azure SQL Server.

@cppwfs / @mminella thinks there will be a migration chore associated with this patch release upgrade.

### PR DESCRIPTION
**Pull request by jvalkeal**
NOTE: task should be switched from 2.3.3-SNAPSHOT when release is available.

- Update spring-cloud-task 2.3.3
- New TASK_SEQ as sequence and migration from table.
- Fixes #4547

### CODE REVIEW

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/db/migration/sqlserver/V4_1__Task_Sequence.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+ * into sequence by taking out a correct seq value.
+ *
+ * @author Janne Valkealahti
+ *
+ */
+public class V4_1__Task_Sequence extends AbstractMigration {
+
+	private static final Logger logger = LoggerFactory.getLogger(R__Hibernate_Sequence.class);
+
+	// Caused by: org.springframework.jdbc.UncategorizedSQLException:
```
Comment (cppwfs) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/db/migration/sqlserver/V4_1__Task_Sequence.java:43]: Do we need this comment?

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/db/migration/sqlserver/V4_1__Task_Sequence.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+		logger.info("About to check if mssql TASK_SEQ needs fix from table to a sequence");
+		try {
+			JdbcTemplate jdbcTemplate = new JdbcTemplate(new SingleConnectionDataSource(context.getConnection(), true));
+			// in case we have old wrongly created table, this command should succeed
+			jdbcTemplate.execute("select 1 from TASK_SEQ");
+			fixSequence = true;
+			logger.info("Looks like we have TASK_SEQ table, initiate fix");
+		}
+		catch (Exception e) {
+			logger.debug("Unable to query TASK_SEQ table, looks like we have a proper sequence", e);
```
Comment (cppwfs) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/db/migration/sqlserver/V4_1__Task_Sequence.java:77]: Reword 
`Unable to query TASK_SEQ table, looks like we have a proper sequence` to `Unable to query TASK_SEQ table, a TASK_SEQ sequence may already exist`

File: spring-cloud-dataflow-server-core/src/main/resources/schemas/sqlserver/V4_1-dataflow.sql [hunk 3]
```diff
@@ -0,0 +1,3 @@
+-- old table should be dropped if exists
+-- drop table TASK_SEQ;
```
Comment (cppwfs) [inline spring-cloud-dataflow-server-core/src/main/resources/schemas/sqlserver/V4_1-dataflow.sql:2]: May want to add a note that states, " Set start with a value of 1 greater than current value in TASK_SEQ table, if TASK_SEQ table has value greater than 1"

**Comment by cppwfs** [comment]
Other than the doc comments above: LGTM


- [ ] 2.7.x -> 2.8x existing apps launched

- [ ] 2.8.x new apps unlaunched

- [ ] mysqldb validation on 2.8.x

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/db/migration/sqlserver/V4_1__Task_Sequence.java (hunk 1 above)
Comment (jvalkeal) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/db/migration/sqlserver/V4_1__Task_Sequence.java:43]: I was a reference why things are done like this. Could remove thought and fix wrong logger class.

**Comment by jvalkeal** [comment]
Closing as superseded by #4568

**Pull request closed without merging**